sleepecg/config.py

```diff
@@ -0,0 +1,98 @@
+# Authors: Florian Hofer
+#
+# License: BSD (3-clause)
+
+"""Functions for getting and setting configuration values."""
+
+import contextlib
+from pathlib import Path
+from typing import Any, Dict, Optional
+
+import yaml
+
+__all__ = [
+    'get_config',
+    'set_config',
+]
+
+_DEFAULT_CONFIG_PATH = Path(__file__).parent / 'config.yml'
+_USER_CONFIG_PATH = Path('~/.sleepecg/config.yml').expanduser()
+
+
+def _read_yaml(path: Path) -> Dict[str, Any]:
+    with open(path) as file:
+        cfg = yaml.safe_load(file)
+        # empty .yml-files are loaded as `None`
+        if cfg is None:
+            return {}
+        return cfg
+
+
+def get_config(key: Optional[str] = None) -> Any:
+    """
+    Read SleepECG preferences from the configuration file.
+
+    For parameters not set in the user configuration file
+    (`~/.sleepecg/config.yml`), this falls back to the default values
+    defined in `site-packages/sleepecg/config.yml`.
+
+    Parameters
+    ----------
+    key : str, optional
+        The configuration key to look for. If `None`, all configuration
+        settings are returned in a dictionary, by default `None`.
+
+    Returns
+    -------
+    typing.Any
+        The configuration value.
+    """
+    config = _read_yaml(_DEFAULT_CONFIG_PATH)
+    with contextlib.suppress(FileNotFoundError):
+        config.update(_read_yaml(_USER_CONFIG_PATH))
+
+    if key is None:
+        return config
+    try:
+        return config[key]
+    except KeyError:
+        options = ', '.join(config)
+        raise ValueError(f'Trying to get invalid config key: {key!r}, possible options: {options}') from None  # noqa: E501
```

Comment:
Why do you `raise from None` here? In `set_config` you are just raising the `ValueError`, which I'd also do here.

Reply:
being inside an `except`-block, this `raise` would let the traceback include the original `KeyError` which might be confusing

Reply:
And what about checking with `if key in config`?

Reply:
So are you going to change it or leave it as is 🤔?

Reply:
right, sorry, fixed!